arbos/l1pricing/l1pricing.go

```diff
@@ -293,13 +390,9 @@ func (ps *L1PricingState) PosterDataCost(message core.Message, sender, poster co
 
 	// Approximate the l1 fee charged for posting this tx's calldata
 	l1Bytes := byteCount + TxFixedCost
-	l1GasPrice, _ := ps.L1BaseFeeEstimateWei()
-	l1BytePrice := arbmath.BigMulByUint(l1GasPrice, params.TxDataNonZeroGasEIP2028)
-	l1Fee := arbmath.BigMulByUint(l1BytePrice, uint64(l1Bytes))
+	pricePerUnit, _ := ps.PricePerUnit()
 
-	// Adjust the price paid by the aggregator's reported improvements due to batching
```

Comment:
Should we add a general discount rate given that on average 1/256 bytes will be 0 in the overall compressed batch?

Reply:
I don't think that is necessary. The pricing algorithm will automatically adjust the price per unit to compensate for any multiplicative discount.